components/script/dom/node.rs

```diff
@@ -1752,6 +1752,16 @@ impl Iterator for TreeIterator {
             }
         }
 
+        if self.shadow_including {
+            if let Some(element) = current.downcast::<Element>() {
+                if let Some(shadow_root) = element.shadow_root() {
+                    self.current = Some(DomRoot::from_ref(shadow_root.upcast::<Node>()));
+                    self.depth += 1;
+                    return Some(current);
+                }
+            }
+        }
```

Comment:
nit: this could be combined with the if block above by doing something like
```rs
// Handle a potential shadow root on the element
if let Some(element) = current.downcast::<Element>() {
    if let Some(shadow_root) = element.shadow_root() {
        if self.shadow_including {
             self.current = Some(DomRoot::from_ref(shadow_root.upcast::<Node>()));
             self.depth += 1;
             return Some(current);
       } else {
            return self.next_skipping_children_impl(current);
       }
    }
}
```
*(seems odd that we skip children of shadow hosts in non shadow-including traversals btw)*